test/e2e/framework/pod/delete.go

```diff
@@ -56,6 +57,20 @@ func DeletePodWithWait(ctx context.Context, c clientset.Interface, pod *v1.Pod)
 	return DeletePodWithWaitByName(ctx, c, pod.GetName(), pod.GetNamespace())
 }
 
+// DeletePodWithWaitByBatch deletes the passed-in pods, waits for the pod to be terminated, return deletion error if exists
+func DeletePodWithWaitByBatch(ctx context.Context, c clientset.Interface, pods []*v1.Pod) {
+	var wg sync.WaitGroup
+	for i, pod := range pods {
+		wg.Add(1)
+		go func(i int, pod *v1.Pod) {
+			defer wg.Done()
+			defer ginkgo.GinkgoRecover()
+			framework.ExpectNoError(DeletePodWithWait(ctx, c, pod), "Error deleting Pod %s", pod.GetName())
+		}(i, pod)
+	}
+	wg.Wait()
```

Comment:
it may be a good improvement to fail test here instead of in each goroutine. This way test will clean up it's Pods more consistently, even if one of them failed to be deleted.